packages/cli/src/modules/insights/insights.service.ts

```diff
@@ -1,361 +1,42 @@
 import type { InsightsSummary } from '@n8n/api-types';
-import { Container, Service } from '@n8n/di';
-import { In } from '@n8n/typeorm';
-import { DateTime } from 'luxon';
-import { Logger } from 'n8n-core';
+import { Service } from '@n8n/di';
 import type { ExecutionLifecycleHooks } from 'n8n-core';
-import {
-	UnexpectedError,
-	type ExecutionStatus,
-	type IRun,
-	type WorkflowExecuteMode,
-} from 'n8n-workflow';
+import type { IRun } from 'n8n-workflow';
 
-import { SharedWorkflow } from '@/databases/entities/shared-workflow';
-import { SharedWorkflowRepository } from '@/databases/repositories/shared-workflow.repository';
 import { OnShutdown } from '@/decorators/on-shutdown';
-import { InsightsMetadata } from '@/modules/insights/database/entities/insights-metadata';
-import { InsightsRaw } from '@/modules/insights/database/entities/insights-raw';
 
 import type { PeriodUnit, TypeUnit } from './database/entities/insights-shared';
 import { NumberToType } from './database/entities/insights-shared';
 import { InsightsByPeriodRepository } from './database/repositories/insights-by-period.repository';
-import { InsightsRawRepository } from './database/repositories/insights-raw.repository';
-import { InsightsConfig } from './insights.config';
-
-const config = Container.get(InsightsConfig);
-
-const shouldSkipStatus: Record<ExecutionStatus, boolean> = {
-	success: false,
-	crashed: false,
-	error: false,
-
-	canceled: true,
-	new: true,
-	running: true,
-	unknown: true,
-	waiting: true,
-};
-
-const shouldSkipMode: Record<WorkflowExecuteMode, boolean> = {
-	cli: false,
-	error: false,
-	retry: false,
-	trigger: false,
-	webhook: false,
-	evaluation: false,
-
-	// sub workflows
-	integrated: true,
-
-	// error workflows
-	internal: true,
-
-	manual: true,
-};
-
-type BufferedInsight = Pick<InsightsRaw, 'type' | 'value' | 'timestamp'> & {
-	workflowId: string;
-	workflowName: string;
-};
+import { InsightsCollectionService } from './insights.collection.service';
+import { InsightsCompactionService } from './insights.compaction.service';
 
 @Service()
 export class InsightsService {
-	private readonly cachedMetadata: Map<string, InsightsMetadata> = new Map();
-
-	private compactInsightsTimer: NodeJS.Timer | undefined;
-
-	private bufferedInsights: Set<BufferedInsight> = new Set();
-
-	private flushInsightsRawBufferTimer: NodeJS.Timeout | undefined;
-
-	private isAsynchronouslySavingInsights = true;
-
-	private flushesInProgress: Set<Promise<void>> = new Set();
-
 	constructor(
-		private readonly sharedWorkflowRepository: SharedWorkflowRepository,
 		private readonly insightsByPeriodRepository: InsightsByPeriodRepository,
-		private readonly insightsRawRepository: InsightsRawRepository,
-		private readonly logger: Logger,
-	) {
-		this.logger = this.logger.scoped('insights');
-	}
+		private readonly insightsCompactionService: InsightsCompactionService,
+		private readonly insightsCollectionService: InsightsCollectionService,
```

Comment:
Nit - can be shorter as it's clear from context and less verbose

```suggestion
		private readonly compactionService: InsightsCompactionService,
		private readonly collectionService: InsightsCollectionService,
```